sympy/calculus/util.py

```diff
@@ -452,7 +483,7 @@ def _check(orig_f, period):
     if symbol not in f.free_symbols:
         return S.Zero
 
-    if isinstance(f, TrigonometricFunction):
+    if hasattr(f, 'period'):
         try:
             period = f.period(symbol)
         except NotImplementedError:
```

Comment:
We can remove `try-except`. Can you find a way of removing it?

Reply:
I believe not, since different functions can implement different algorithm for its ``period`` method. See ``period`` method of ``sympy.functions.elementary.TrigonometricFunction``.